lightning-invoice/src/payment.rs

```diff
@@ -1860,6 +1842,8 @@ mod tests {
 
 					// Since the path is reversed, the last element in our iteration is the first
 					// hop.
+					let mut locked_scorer = self.scorer.lock();
+					let scorer = AccountForInFlightHtlcs::new(locked_scorer.deref_mut(), inflight_htlc_map.clone());
```

Comment:
It's a little strange that we are testing `AccountForInFlightHtlcs` this way rather through `DefaultRouter`. Then again, seems doing the latter would require using the actual `find_route` code, which wouldn't lend to a simple test setup. Fine with keeping this as is, but wanted to raise this in case others have any better ideas.